packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -214,3 +241,96 @@ export const LocationsViewTable = (): JSX.Element => {
     />
   );
 };
+
+export const LocationDetailViewTable = (): JSX.Element => {
+  const [{ history, location }, handleUpdateState] = useControl({
+    type: 'NAVIGATE',
+  });
+
+  const [{ data, isLoading }, handleList] = useAction({
+    type: 'LIST_LOCATION_ITEMS',
+  });
+
+  const { prefix: initialPrefix } = location
+    ? parseLocationAccess(location)
+    : ({} as LocationData);
+
+  const prefix =
+    history.length === 1 ? initialPrefix : history[history.length - 1];
+
+  const hasItems = !!data.result?.length;
+  const shouldReset = hasItems && !location && !history.length;
+
+  React.useEffect(() => {
+    if (shouldReset) {
+      handleList({ prefix: '', options: { reset: true } });
+    }
+
+    if (typeof prefix !== 'string') return;
+
+    handleList({ prefix, options: { pageSize: 1000, refresh: true } });
+  }, [handleList, prefix, shouldReset]);
+
+  // @TODO: This should be it's own component instead of using `useCallback`
+  const renderRowItem: RenderRowItem<LocationItem> = React.useCallback(
+    (row, index) => {
+      const parseTableData = (
+        row: LocationItem,
+        column: Column<LocationItem>
+      ) => {
+        if (
+          row.type === 'FILE' &&
+          // @ts-ignore @TODO fix this ts error: This comparison appears to be unintentional because the types '"key" | "type"' and '"lastModified"' have no overlap.
+          column.key === 'lastModified' &&
+          row[column.key]
+        ) {
+          return new Date(row[column.key]).toLocaleString();
+        } else {
+          return row[column.key];
+        }
+      };
+
+      return (
+        <TableRow key={index}>
+          {LOCATION_DETAIL_VIEW_COLUMNS.map((column) => {
+            if (row.key === prefix) {
+              // Don't render the current prefix as a row
+              return null;
+            }
+
+            return (
+              <TableData key={`${index}-${column.header}`}>
+                {column.key === 'key' && row.type === 'FOLDER' ? (
+                  <button
+                    onClick={() => {
+                      handleUpdateState({
+                        type: 'NAVIGATE',
+                        prefix: row.key,
+                      });
+                    }}
+                    key={`${index}-${row.key}`}
+                  >
+                    {row.key}
+                  </button>
+                ) : (
+                  <>{parseTableData(row, column)}</>
+                )}
+              </TableData>
+            );
+          })}
+        </TableRow>
+      );
+    },
+    [handleUpdateState, prefix]
+  );
+
+  return isLoading && !hasItems ? (
+    <span>loading...</span>
+  ) : (
+    <TableControl
+      columns={LOCATION_DETAIL_VIEW_COLUMNS}
+      data={data.result}
+      renderRowItem={renderRowItem}
+    />
+  );
```

Comment:
This is the new code I added based off of Caleb's changes. Everything else `LocationDetailViewTable` was copied and pasted from `LocationDetailView`